go/vt/vtgate/planbuilder/testdata/aggr_cases.json

```diff
@@ -5879,19 +5784,18 @@
     }
   },
   {
-    "comment": "aggregation on top of derived table with limit",
-    "query": "select count(val2), sum(val2) from (select id, val2 from user where val2 is null limit 2) as x",
+    "comment": "aggregation on union",
+    "query": "select sum(col) from (select col from user union all select col from unsharded) t",
     "plan": {
       "QueryType": "SELECT",
-      "Original": "select count(val2), sum(val2) from (select id, val2 from user where val2 is null limit 2) as x",
+      "Original": "select sum(col) from (select col from user union all select col from unsharded) t",
       "Instructions": {
         "OperatorType": "Aggregate",
         "Variant": "Scalar",
-        "Aggregates": "count(0) AS count(val2), sum(1) AS sum(val2)",
+        "Aggregates": "sum(0) AS sum(col)",
```

Comment:
hmm... we should be able to push this down, right? we need to think about pushing aggregation into an UNION. don't think the original paper we based this on covered this case, but I don't see a reason we can extend that logic to cover UNION as well. not necessary in this PR, but sometime in the future

Reply:
sounds good.
looks like the original test got modified, we should keep that as well